### PR TITLE
Add more detail to yaml test descriptions

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
### Description 
Added more detail to yaml test descriptions. These changes should be added as they are part of the effort write all of the constraints from the constraint tracker.

### Changes
- Modified yaml tests to add more detail to the test description. 
- Modified `feature-external-constraints.xml` to change `has-data-flow-diagram` level from `ERROR` to `WARNING` (correction).

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
~~- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~~
~~- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~~

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
             <expect id="has-data-flow" target="system-characteristics" test="data-flow" level="WARNING">
                 <message>A FedRAMP SSP must include a data flow section.</message>
             </expect>
-            <expect id="has-data-flow-diagram" target="system-characteristics/data-flow" test="diagram" level="ERROR">
+            <expect id="has-data-flow-diagram" target="system-characteristics/data-flow" test="diagram" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:116]: Sorry, just want to understand: is this change based on reviewing the constraint tracker, was something else miscommunicated here? This is the only change I noticed when skimming that is not directly related to better YAML descriptions.

Otherwise, those descriptions look helpful!

Comment (aj-stein-gsa) [reply src/validations/constraints/fedramp-external-constraints.xml:116]: Sorry, just to be clear, I am talking about the `@level="WARNING"` versus `"ERROR"` bit, as that changes something other than description.

If that needs to change, that is fine, but I was multi-tasking and need to look at the tracker. 😄

Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml:116]: > Sorry, just to be clear, I am talking about the `@level="WARNING"` versus `"ERROR"` bit, as that changes something other than description.
> 
> If that needs to change, that is fine, but I was multi-tasking and need to look at the tracker. 😄

Oh sorry, I misread that then. That change regarding the `@level="WARNING"` was because I misread the constraint tracker. Sloppy work on my part.

**Review by aj-stein-gsa — APPROVED** [review]
